src/Fullcount.sol

```diff
@@ -561,38 +605,41 @@ contract Fullcount is EIP712 {
 
         Session storage session = SessionState[sessionID];
 
-        require(_isTokenOwner(session.batterNFT.nftAddress, session.batterNFT.tokenID), "Fullcount.revealSwing: msg.sender is not batter NFT owner");
+        require(
+            _isTokenOwner(session.batter.nft.nftAddress, session.batter.nft.tokenID),
```

Comment:
Maybe better to return the address of the owner and do an explicit `msg.sender == _tokenOwner(...)`. That can be reused to get rid of `sessionStarter`.